### PR TITLE
27 | create UniversalModal

### PR DESCRIPTION
**Pull request by NatashaLifer**
Add component UniversalModal
https://github.com/BUI-studies/yidlorama/issues/27

### CODE REVIEW

File: src/pages/Home.tsx
Comment (NatashaLifer) [inline src/pages/Home.tsx:1]: Nothing has changed in the file.

**Review by BUI-studies — CHANGES_REQUESTED** [review]
to complicated. start from scratch and write a simple modal window

File: src/components/UniversalModal/UniversalModal.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import classes from './UniversalModal.module.scss'
+
+type ModalProps = {
+    children: ReactNode
+    title: string
+    visible: boolean
+    setVisible: Dispatch<SetStateAction<boolean>>
+}
+
+const UniversalModal:FC<ModalProps> = (props) => {
```
Comment (BUI-studies) [inline src/components/UniversalModal/UniversalModal.tsx:12]:
```suggestion
const UniversalModal:FC<ModalProps> = ({ children, title, visible, setVisible  }) => {
```

File: src/components/UniversalModal/UniversalModal.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const { children, title, visible, setVisible  } = props
```
Comment (BUI-studies) [inline src/components/UniversalModal/UniversalModal.tsx:13]:
```suggestion

```

File: src/components/UniversalModal/UniversalModal.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    
+    const rootClasses = [classes.modal]
```
Comment (BUI-studies) [inline src/components/UniversalModal/UniversalModal.tsx:15]:
```suggestion
    const rootClasses = [classes.modal, visible ? rootClasses.push(classes.active) : '']
```

File: src/components/UniversalModal/UniversalModal.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    visible ? rootClasses.push(classes.active) : null
```
Comment (BUI-studies) [inline src/components/UniversalModal/UniversalModal.tsx:16]:
```suggestion
```

File: src/pages/Home.tsx
Comment (BUI-studies) [inline src/pages/Home.tsx:1]: revert back this file using git, so it doesn't appear in the PR.

**Review by BUI-studies — CHANGES_REQUESTED** [review]

**Review by BUI-studies — APPROVED** [review]

**Review by yamnyk — APPROVED** [review]

**Pull request merged**